### DISCUSSION TITLE
Using business process framework with RAG to peform a guided interview with a user, which might involve one or more human in the loop

### DISCUSSION THREAD

**Discussion by radrad** [microsoft/semantic-kernel#10023] [Q&A]
How can I use business process framework to perform an AI guided interview with 20+ questions that are given in certain order with some flexibility to ask sub questions based on user responses and some might be created to bring more insights. 
How can we have 2 humans in a loop, one of them might be an interviewee and another migh supervise two AI agents (i.e. can take over some part of the conversation that one of the AI agent is doing). One AI agent might be an HR agent and another would be Some domain expert AI agent. So they both would ask questions from a question knowledgebase (HR would have one set and domain expert agent would have another) where initially HR agent would ask more questions and later giving more time for domain expert to ask questions. The goal is to collect employmnet profile or behavioural profile. How about this scenario involving more than one human in the loop.
I would like some human who might be observing the interview to take over the role of AI agent and ask questions when there is a need or to augment some of the questions for AI agent so the messages are simply added on behalf of AI agent. There could also be some quality control where an analyst observer can collect some observations/note about the quality of conversations.

**Comment by sophialagerkranspandey** [comment]
I'm tagging @alliscode to dive into this further. Thanks!

**Comment by markwallace-microsoft** [comment]
I am cleaning up old unanswered discussions
